src/repository.rs

```diff
@@ -403,6 +411,37 @@ impl Repository {
     self.inner.set_head(&refname)?;
     Ok(())
   }
+
+  #[napi]
+  /// Extract a signature from an object identified by its ID.
+  ///
+  /// This method can be used for any object that may be signed, such as commits or tags.
+  ///
+  /// @category Repository/Methods
+  ///
+  /// @signature
+  /// ```ts
+  /// class Repository {
+  ///   extractSignature(oid: string): { signature: string | null, signedData: string } | null;
+  /// }
+  /// ```
+  ///
+  /// @returns An object containing the signature and signed data if the object is signed,
```

Comment:
There's a missing comment about the parameters here, could you please add it?
```suggestion
  /// @param {string} oid - The id of git object.
  /// @returns An object containing the signature and signed data if the object is signed,
```

Reply:
```suggestion
  /// @param {string} oid - Object ID (SHA1) of the signed object to extract the signature from.
  /// @returns An object containing the signature and signed data if the object is signed,
```
I suggest "@param {string} oid - Object ID (SHA1) of the signed object to extract the signature from" as it provides more context about format (SHA1) and purpose (for signature extraction), which helps users understand exactly what to provide.

Reply:
looks good to me 👍